From 49192654fd2471ab163e24874027185a50a41846 Mon Sep 17 00:00:00 2001
From: Hannah von Reth <vonreth@kde.org>
Date: Sat, 24 Feb 2018 12:09:24 +0100
Subject: [PATCH 2/2] Use Locale::name() instead of Locale::bcp47Name()

Summary:
bcp47Name() is encoded like "en-US" the dictionary name on the other hand
is "en_US" so lets use name().

Reviewers: dfaure, mlaurent, kfunk, vkrause

Reviewed By: dfaure

Subscribers: apol, #frameworks

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D10741
---
 autotests/test_settings.cpp | 6 +++---
 src/core/settings.cpp       | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/autotests/test_settings.cpp b/autotests/test_settings.cpp
index 4757547..e80b87b 100644
--- a/autotests/test_settings.cpp
+++ b/autotests/test_settings.cpp
@@ -82,7 +82,7 @@ void SonnetSettingsTest::testSpellerAPIChangeSaves()
     // Default Language
     QString defLang = speller.defaultLanguage();
     QString settingsLang = settings.value(QStringLiteral("defaultLanguage"),
-                                          QLocale::system().bcp47Name()).toString();
+                                          QLocale::system().name()).toString();
     QCOMPARE(defLang, settingsLang);
 
     QStringList langs = speller.availableLanguages();
@@ -90,7 +90,7 @@ void SonnetSettingsTest::testSpellerAPIChangeSaves()
         if (langs[i] != defLang) {
             speller.setDefaultLanguage(langs[i]);
             settingsLang = settings.value(QStringLiteral("defaultLanguage"),
-                                          QLocale::system().bcp47Name()).toString();
+                                          QLocale::system().name()).toString();
             QCOMPARE(settingsLang, langs[i]);
             QCOMPARE(speller.defaultLanguage(), langs[i]);
             break;
@@ -100,7 +100,7 @@ void SonnetSettingsTest::testSpellerAPIChangeSaves()
     speller.setDefaultLanguage(defLang);
     settingsLang
         = settings.value(QStringLiteral("defaultLanguage"),
-                         QLocale::system().bcp47Name()).toString();
+                         QLocale::system().name()).toString();
     QCOMPARE(settingsLang, defLang);
     QCOMPARE(speller.defaultLanguage(), defLang);
 
diff --git a/src/core/settings.cpp b/src/core/settings.cpp
index 4793cd1..6e823bf 100644
--- a/src/core/settings.cpp
+++ b/src/core/settings.cpp
@@ -271,7 +271,7 @@ void Settings::restore()
     QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
     d->defaultClient = settings.value(QStringLiteral("defaultClient"), QString()).toString();
     d->defaultLanguage = settings.value(QStringLiteral("defaultLanguage"),
-                                        QLocale::system().bcp47Name()).toString();
+                                        QLocale::system().name()).toString();
 
     //same defaults are in the default filter (filter.cpp)
     d->checkUppercase = settings.value(QStringLiteral("checkUppercase"), true).toBool();
-- 
2.16.1.windows.4

